chingu-voyages/v51-tier3-team-33
# Olga/feature/add group frontend

# Description

Created a form to add new group.

## Main changes explained:

- Created UI for the add new group form
- Implemented  functionality to add members to group
- Added api calls to handle group creation in database
- Also fixed styling of the image on the login screen (not related to the purpose of this PR though)

## How to test:

1. Check into current branch
2. Do `pnpm install` and `pnpm run dev` to run this PR locally
3. Go to `/login` 
4. Login with google
5. In the database (using MongoDB Compass) find your user and manually add friends to it (add other IDs from 'users' collection). This is needed because you can only add your friends to this group. Here is what it should look like:

<img width="700" alt="Screenshot 2024-10-08 at 10 17 52 AM" src="https://github.com/user-attachments/assets/d05efe10-03af-4a0f-8a79-a13c3b569323">

6. Go to `/home` and click 'Add Group' button on the sidebar. Fill in this form and add friends.

![Screenshot 2024-10-08 at 10 21 27 AM](https://github.com/user-attachments/assets/6d96a8f7-a56e-4e83-9207-fa87c349938c)

7. Verify that 'Cancel' button resets all form fields
8. Verify that 'Create' button creates a group and resets the form. To check that a group has been created, check 'Groups' collection in MongoDB Compass. Your group should be there.
9. Finally reload the page and check that your newly created group is displayed in the sidebar under 'You groups':
 
![Screenshot 2024-10-08 at 10 27 34 AM](https://github.com/user-attachments/assets/4bdd1550-f530-4adc-afba-49dd5bafd1e5)



## Note:

The' add new group form' is not optimized for mobile screen. This is something to be done later.
